KarishmaThaladi_final_assessment.py

```diff
@@ -0,0 +1,211 @@
+in front of the id.
+
+New methods ‘add_new_grade’ and ‘get_course_grades’ should be added.
+You can use  it as a skeleton code for your classes OR adjust it and create your own.
+
+SEE NOTES BELOW
+
+"""
+import random
+
+class CFGStudent:
+
+    def __init__(self, name, surname, age, email):
+        self.name = name
+        self.surname = surname
+        self.age = age
+        self.email = email
+        self.student_id = random.randint(1000,10000)
+
+    @staticmethod
+    def generate_id():
+        'your code goes here'
+        'create a random new id, which is any number between 1,000 and 10,000'
+        'return id as a string'
+        "Example '8754' "
+        return self.student_id
+
+    def get_id(self):
+        return '{}'.format(self.student_id)
+        'fetch student id'
+
+    def get_fullname(self):
+        "Expected outcome should be 'Name Surname' "
+        return '{} {}'.format(self.name, self.surname)
+
+
+class NanoStudent(CFGStudent):
+
+    def __init__(self, specialization, name, surname, age, email, course_grades=None, students=None ):
+        super().__init__(name, surname, age, email)
+        self.specialization = specialization
+
+        if students is None:
+            self.students = []
+        else:
+            self.students = students
+
+        if course_grades is None:
+            self.course_grades = []
+        else:
+            self.course_grades = course_grades
+
+    @staticmethod
+    def generate_id():
+        'your code goes here'
+        'create a random new id, which is a word NANO followed by any number between 1,000 and 10,000'
+        'return id as a string'
+        "Example 'NANO1234' "
+        return 'NANO'+str(random.randint(1000, 10000))
+
+    def add_new_grade(self, new_grade):
+        'your code goes here'
+        for course_grades in self.course_grades:
+            self.new_grade = new_grade
+        'update course_grades container'
+        "Example: pass in a task name and its grade, so that both are added to course_grades"
+
+    def get_course_grades(self):
+        'your code goes here'
+        'fetch course grades container and its values'
+        for student in self.students:
+            print('--> ' + student.name() + student.surname() + '--> ' + student.course_grades)
+
+
+############################################
+
+# Example run
+
+s = CFGStudent('Anna', 'Smith', 18, 'anna@mail.com')  # do not pass ID, it should be generated automatically
+print(s.get_fullname())
+# returns 'Anna Smith'
+print(s.student_id)
+# returns '3868' or some other random number
+
+cfg_s = NanoStudent('Software', name='Mia', surname='Papandopulu', age=20, email='mia@mail.com')
+print(cfg_s.get_fullname())
+# returns 'Mia Papandopulu'
+print(cfg_s.get_id())
+# returns 'NANO6180' or some other random number
+
+cfg_s.add_new_grade('theory', 95)
+cfg_s.add_new_grade('project', 78)
+print(cfg_s.get_course_grades())
+# returns {'theory': 95, 'project': 78}
+

```

Comment:
Very good solution. The only thing that doesn't work is the add_new_grade method. Perhaps have a look at this again.